controllers/storagecluster/provider_server.go

```diff
@@ -146,11 +146,15 @@ func (o *ocsProviderServer) createDeployment(r *StorageClusterReconciler, instan
 
 func (o *ocsProviderServer) createService(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
 
-	if instance.Spec.ProviderAPIServerServiceType != "" && instance.Spec.ProviderAPIServerServiceType != corev1.ServiceTypeNodePort &&
-		instance.Spec.ProviderAPIServerServiceType != corev1.ServiceTypeLoadBalancer {
-		err := fmt.Errorf("providerAPIServer only supports service of type %s and %s", corev1.ServiceTypeNodePort, corev1.ServiceTypeLoadBalancer)
-		r.Log.Error(err, "Failed to create/update service, Requested ServiceType is", "ServiceType", instance.Spec.ProviderAPIServerServiceType)
-		return reconcile.Result{}, err
+	if instance.Spec.ProviderAPIServerServiceType != "" {
+		switch instance.Spec.ProviderAPIServerServiceType {
+		case corev1.ServiceTypeClusterIP, corev1.ServiceTypeLoadBalancer, corev1.ServiceTypeNodePort:
+		default:
+			err := fmt.Errorf("providerAPIServer only supports service of type %s, %s and %s", corev1.ServiceTypeNodePort, corev1.ServiceTypeLoadBalancer, corev1.ServiceTypeClusterIP)
```

Comment:
Can you pls divide it in 2 lines for better readability?

```
err := fmt.Errorf("providerAPIServer only supports service of type %s, %s and %s", 
    corev1.ServiceTypeNodePort, corev1.ServiceTypeLoadBalancer, corev1.ServiceTypeClusterIP)
```

Reply:
done